CryptomatorFileProvider/FileProviderItem.swift

```diff
@@ -157,9 +157,21 @@ public class FileProviderItem: NSObject, NSFileProviderItem {
 		return metadata.tagData
 	}
 
-	/// Workaround to access the `isUploading` and `uploadingError` property in the `NSExtensionFileProviderActionActivationRule`
+	/**
+	 Dictionary to add state information to the item. Entries are accessible to
+	 user interaction predicates via the `Info.plist` of the FileProviderExtensionUI
+
+	 Used as workaround to access the item properties:
+	    - `isUploading`
+	    - `isDownloaded`
+	    - `isDownloading`
+	  Furthermore adds information about existing upload errors via `hasUploadError` and if the item is a folder via `isFolder`.
+	 */
 	public var userInfo: [AnyHashable: Any]? {
 		return ["isUploading": isUploading,
-		        "hasUploadError": uploadingError != nil]
+		        "hasUploadError": uploadingError != nil,
+		        "isDownloaded": isDownloaded,
+		        "isDownloading": isDownloading,
+		        "isFolder": typeIdentifier == kUTTypeFolder as String]
```

Comment:
Since we've noticed in a call that this workaround is potentially confusing: Are these still mutually exclusive? Would it be possible to create an enum for it? Maybe something like `state`?

Something like that would probably be needed for the `NSExtensionFileProviderActionActivationRule`: https://stackoverflow.com/a/32299215/1759462

Reply:
I am a bit unsure if using an enum has an advantage here. The workaround is actually just to make the `NSFileProviderItem` properties that are of interest to us available - this _should_ actually work directly via `$fileproviderItem.<propertyName>`. 
I also think that using `CustomStringConvertible` doesn't do anything in our case, since we work with [predicate format strings](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/Predicates/Articles/pSyntax.html#//apple_ref/doc/uid/TP40001795) in the `Info.plist`.

But we could, to avoid logic errors in the predicate format strings in the `Info.plist`, move the logic into the creation of the `userInfo` dictionary. This would make the current "predicate logic" unit-testable and we would then have a `userInfo` that would look like this:

```swift
["enableRetryWaitingUploadAction": Bool,
 "enableRetryFailedUploadAction": Bool,
 "enableEvictFileFromCacheAction": Bool]
```

Because I'm not sure if we should actually decide that the activation of the actions should be mutually exclusive - what happens as soon as we introduce an action that should additionally be activated for some items that are cached or similar?

Reply:
I like your suggestion and it would improve readability and maintainability. Enum indeed wouldn't improve anything. 😄

Reply:
Resolved by b31862f2300261d7c7ac08083d37ebee9dcff0d5.